nimbus/src/enrollment.rs

```diff
@@ -327,6 +287,28 @@ impl ExperimentEnrollment {
         })
     }
 
+    /// Reset identifiers on application-level telemetry opt-out.
+    ///
+    /// Each experiment enrollment record contains a unique `enrollment_id`. When the user
+    /// opts out of application-level telemetry, we reset each such id to a special nil value,
+    /// creating a clean break between data sent before the opt-out and any data that might be
+    /// submitted about these enrollments in future.
+    ///
+    /// We also move any enrolled experiments to the "disqualified" state, since their
+    /// further partipation in the experiment may skew the data if they later re-enable
+    /// telemetry.
+    ///
+    /// Note that this does not return any change events; since the user has opted out
+    /// of telemetry, we wouldn't be able to submit said events anyway.
```

Comment:
Hrm, according to https://mana.mozilla.org/wiki/pages/viewpage.action?spaceKey=FJT&title=Nimbus+Engineering#NimbusEngineering-Events we're supposed to emit an `Unenrollment` event when telemetry is disabled. My impression was that, since the user has disabled telemetry, we would not try to send any further events. @travis79 do you have any insight into the correct behaviour here?

Reply:
If telemetry is disabled, it won't make it out. Glean drops all current telemetry and sends a deletion_request ping.  What we might want to do in this case is include the enrollment_id in the deletion_request ping as a sort of unenrollment event.  I'm not certain what we can or would do with the ids (or the event for that matter) since the deletion_request ping will eventually erase all unaggregated data for that user anyway.